utils/s3/client.cc

```diff
@@ -89,6 +90,19 @@ future<> ignore_reply(const http::reply& rep, input_stream<char>&& in_) {
 client::client(std::string host, endpoint_config_ptr cfg, semaphore& mem, global_factory gf, private_tag, std::unique_ptr<aws::retry_strategy> rs)
         : _host(std::move(host))
         , _cfg(std::move(cfg))
+        , _creds_invalidation_timer([this] {
+            s3l.info("Invalidating creds");
+            _credentials = {};
+        })
+        , _creds_update_timer([this] {
+            (void)[this]->future<> {
```

Comment:
I love this one. It took me a while to parse wtf `this->future<>()` invocation means and why the hell it's wrapped this weird way :exploding_head:

Reply:
hope it is more readable now